<launch>

    <!-- map_file -->
    <!-- <arg name="map_file" default="/home/jing/cartobot_ws/src/cartographer_ros/cartographer_ros/maps/PME_3F.yaml"/> -->
    <!-- <arg name="map_file" default="/home/jing/Desktop/0724.yaml"/> -->
    <arg name="map_file" default="$(find diff_navigation)/maps/0724_newnew.yaml"/>

    
    <!-- map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" >
        <remap from="/map" to ="/carto_map" />
    </node>

    <node name="odom_base_footprint" pkg="tf2_ros" type="static_transform_publisher" args="0 0 0 0 0 0 1 base_footprint base_link"/>
    
    <!-- gloabol planner -->
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <remap from="/map" to="/carto_map" />
        <param name="base_global_planner" value="global_planner/GlobalPlanner"/>
        <rosparam file="$(find diff_navigation)/params/global_costmap_param.yaml" command="load" />
        <rosparam file="$(find diff_navigation)/params/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find diff_navigation)/params/global_planner_param.yaml" command="load" />
    </node>

    <!-- differential-controller -->
    <!-- <node pkg="diff_navigation" type="diff_controller" name="diff_controller" output="screen"> -->
    <!-- <node pkg="diff_navigation" type="diff_controller_plus" name="diff_controller_plus" output="screen">
        <rosparam file="$(find diff_navigation)/params/diff_controller_param.yaml" command="load" />
    </node> -->

    <!-- <node pkg="diff_navigation" type="regulated_pure_pursuit" name="regulated_pure_pursuit" output="screen">
        <param name="if_allow_reversing" value="true" />
        <remap from="/ekf_pose" to ="/carto_pose" />
        <rosparam file="$(find diff_navigation)/params/regulated_pure_pursuit_test.yaml" command="load" />
    </node> -->

    <node pkg="diff_navigation" type="diff_controller_avoidance" name="diff_controller_avoidance" output="screen">
        <param name="if_avoidance_enable" value="false" />
        <param name="if_allow_reversing" value="false" />
        <rosparam file="$(find diff_navigation)/params/real_diff_controller_avoidance_param.yaml" command="load" />
    </node>


</launch>
